src/js_realm_object.hpp

```diff
@@ -235,14 +236,16 @@ std::vector<String<T>> RealmObjectClass<T>::get_property_names(ContextType ctx,
     auto realm_object = get_internal<T, RealmObjectClass<T>>(object);
     auto &object_schema = realm_object->get_object_schema();
 
+    // When parsing schema objects from JS, we always ensure an alias is set. If the user didn't define
+    // one, the name property will be used as the alias.
```

Comment:
Doesn't this disagree with the logic in object store validation which checks that there are no conflicts between aliases and names?

Reply:
Yes, this comment isn't correct anymore. I'll update it.